extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -554,6 +563,7 @@ class CornerstoneViewportService implements IViewportService {
       viewportInfo.getViewportId()
     );
 
+    this.hangingProtocolService.applyCustomViewportStore(publicViewportOptions, priorViewportOptions, viewport);
```

Comment:
I don't like the idea of adding a store for the HangingProtocolService, it gets out of hand immediately e.g,. currently displaying extensionA's viewport, then drag and drop a displaySet which will cause extensionB's viewport to mount. You can achieve the store functionality by an external service which will listen to enabledElement or so.

Reply:
No actually, you can't achieve it because it gets called on both the old and the new settings values.  If you want to have access to it, then you need to build it in.  Since the hanging protocol service owns the viewport settings, that is the appropriate place to build it in.  If they don't get used, then the calls are just no-ops, but if you need it, I don't see a good way of getting access other than constant forking the OHIF code.

The general pattern I'm trying to make in changes is to provide hooks into various places to allow changes to be made, without otherwise making deep changes, so that it becomes possible to inject certain types of changes.  Yes, if the underlying code is re-implemented, and the hook isn't applicable, then that change doesn't apply any longer (eg if the viewport changes such that setPan/setZoom no longer apply), but that is the nature of hooks, that they depend on exactly the spot they get added to.